zaza/openstack/charm_tests/octavia/tests.py

```diff
@@ -99,11 +100,77 @@ def revoke_role_current_user(keystone_client, keystone_session, role_name,
 class CharmOperationTest(test_utils.OpenStackBaseTest):
     """Charm operation tests."""
 
+    application_name = "octavia"
+    states = {"octavia": {
+        "workload-status": "active",
+        "workload-status-message": "Unit is ready"}}
+
     @classmethod
     def setUpClass(cls):
         """Run class setup for running Octavia charm operation tests."""
         super(CharmOperationTest, cls).setUpClass()
 
+    def get_leaders_and_non_leaders(self):
```

Comment:
You copy-pasted this function from `mysql/tests.py`. Please do not duplicate code. Move it to a central place instead. Thanks!

Reply:
move to zaza/openstack/utilities/generic.py now